### DISCUSSION TITLE
SSH URL for cloning a repo is not presented as an option to logged-out users

### DISCUSSION THREAD

**Discussion by confluence** [community/community#18791] [Other Feature Feedback, Questions, & Ideas]
I think that this behaviour may have been introduced when the new GitHub CLI URL option was added (although perhaps I just haven't ever noticed it before). This has confused at least one of my colleagues, who did not realise that he was not logged in, and thought that the SSH option had been deprecated in favour of the GH CLI.

More broadly, I think that this behaviour is confusing and unexpected, because whether I am logged in or not in whatever browser I am using to view a repository page is unrelated to whether I have the correct keys, etc., set up in the terminal where I want to paste the command. The browser may not be my usual browser, the terminal may be a remote session on a completely different computer, and there are many possible reasons why it may not be convenient for me to log in in the browser.

I understand that some users may not realise that the SSH URL will not work out of the box like the others -- but would it not be possible to add a more detailed explanation to the option text to clarify this? The user being logged in doesn't guarantee on its own that the URL will work.

Is this new behaviour, or has it always been like this?

**Comment by rudimeier** [comment]
This issue is now even more annoying, since we are forced to use two-factor authentication.

In short:  I always want to clone with ssh URLs only, regardless if I am logged in or not, as it was possible in past.
